results-processor/processor_test.py

```diff
@@ -232,3 +233,42 @@ def test_download_content_disposition(self):
             # artifact_test.zip as the filename.
             path = p._download_single(self.url + '/download/attachment')
             self.assertTrue(path.endswith('.zip'))
+
+
+class ProcessorAPIServerTest(unittest.TestCase):
+    def setUp(self):
+        self.server, self.url = test_util.start_server(True)
+
+    def tearDown(self):
+        if self.server.poll() is None:
+            self.server.kill()
+
+    def test_update_status(self):
+        with Processor() as p:
+            p._auth = AUTH
+            p.report.update_metadata(
+                revision='21917b36553562d21c14fe086756a57cbe8a381b')
+            p.update_status(
+                run_id='12345', stage='INVALID',
+                error='Sample error', callback_url=self.url)
+        self.server.terminate()
+        _, err = self.server.communicate()
+        self.assertEqual(
+            err,
+            b'{"id": 12345, "stage": "INVALID", "error": "Sample error", '
+            b'"full_revision_hash": "21917b36553562d21c14fe086756a57cbe8a381b"}\n')
+
+    def test_create_run(self):
+        api = self.url + '/api/results/create'
+        with Processor() as p:
+            p._auth = AUTH
+            p.report.update_metadata(
+                browser_name='chrome',
+                browser_version='70',
+                os_name='Linux',
+                revision='21917b36553562d21c14fe086756a57cbe8a381b')
+            p.create_run('12345', '', 'blade-runner', callback_url=api)
+            self.assertEqual(p.test_run_id, 12345)
```

Comment:
It's not very clear to me what we're testing here. We call `create_run` and then check the value of a field in the Processor object we have. It may be worth a brief comment noting that if `create_run` succeeds it sets this field.

(Side-bar for my education; does this impl a Processor object handles exactly one run? Or one run at a time?)

Reply:
This is a white-box test, replying on the fact that we set this field (`p.test_run_id`) based on the response from `/api/results/create`, which in turn uses the ID provided by the caller of that API (`p.create_run`). This essentially tests that `create_run` passes the run ID correctly to the server. I'll write a comment.

(Yes, one run at a time / per request.)